src/helpers.rs

```diff
@@ -285,10 +290,53 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
 
             // We should never get to a primitive, but always short-circuit somewhere above.
-            fn visit_primitive(&mut self, _v: MPlaceTy<'tcx, Tag>) -> InterpResult<'tcx>
-            {
+            fn visit_primitive(&mut self, _v: MPlaceTy<'tcx, Tag>) -> InterpResult<'tcx> {
                 bug!("we should always short-circuit before coming to a primitive")
             }
         }
     }
+
+    fn write_c_ints(
```

Comment:
I have no idea what this function does, please add a doc comment.

Reply:
Looking at ow you are using this, I am not convinced this is the right abstraction.

What *does* seem reasonable is "here's a list of `ImmTy`, write them as if they are a packed struct". But having libc type-names hard-coded in the same abstraction strikes me as too specific.